Developed in-house reflecting community issue.
http://redmine.lighttpd.net/issues/1239
Solaris-specific, currently ignored by community.

--- src/network.c	2013-11-05 09:41:41.841723296 -0800
+++ src/network.c	2013-11-05 09:40:25.359053258 -0800
@@ -251,6 +251,10 @@
 		log_error_write(srv, __FILE__, __LINE__, "ss", "socketsockopt(SO_REUSEADDR) failed:", strerror(errno));
 		goto error_free_socket;
 	}
+        if (setsockopt(srv_socket->fd, IPPROTO_TCP, TCP_NODELAY, &val, sizeof(val)) < 0) {
+                log_error_write(srv, __FILE__, __LINE__, "ss", "socketsockopt(TCP_NODELAY) failed:", strerror(errno));
+                goto error_free_socket;
+        }
 
 	switch(srv_socket->addr.plain.sa_family) {
 #ifdef HAVE_IPV6
